src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PublisherEventsPersistor.java

```diff
@@ -41,33 +42,35 @@ public PublisherEventsPersistor(final Persistable persistable) {
         this.persistable = persistable;
     }
 
-    public void persistAction(final Action action) {
+    public void persistAction(final List<Action> actionsPublished) {
         long timestamp = Instant.now().toEpochMilli();
-        LOG.debug("Action: [{}] published to persistor publisher.", action.name());
-        PerformanceAnalyzerApp.RCA_RUNTIME_METRICS_AGGREGATOR.updateStat(
-                RcaRuntimeMetrics.ACTIONS_PUBLISHED, action.name(), 1);
-        if (action.impactedNodes() != null) {
-            final String nodeIds = action.impactedNodes().stream()
-                                           .map(n -> n.getNodeId().toString())
-                                           .collect(Collectors.joining(",", "{", "}"));
-            final String nodeIps = action.impactedNodes().stream()
-                                           .map(n -> n.getHostAddress().toString())
-                                           .collect(Collectors.joining(",", "{", "}"));
-            final PersistedAction actionsSummary = new PersistedAction();
-            actionsSummary.setActionName(action.name());
-            actionsSummary.setNodeIds(nodeIds);
-            actionsSummary.setNodeIps(nodeIps);
-            actionsSummary.setActionable(action.isActionable());
-            actionsSummary.setCoolOffPeriod(action.coolOffPeriodInMillis());
-            actionsSummary.setMuted(action.isMuted());
-            actionsSummary.setSummary(action.summary());
-            actionsSummary.setTimestamp(timestamp);
-            try {
-                persistable.write(actionsSummary);
-            } catch (Exception e) {
-                LOG.error("Unable to write publisher events to sqlite", e);
-                PerformanceAnalyzerApp.ERRORS_AND_EXCEPTIONS_AGGREGATOR.updateStat(
-                        ExceptionsAndErrors.EXCEPTION_IN_PERSIST, action.name(), 1);
+        for (Action action : actionsPublished) {
+            LOG.debug("Action: [{}] published to persistor publisher.", action.name());
+            PerformanceAnalyzerApp.RCA_RUNTIME_METRICS_AGGREGATOR.updateStat(
+                    RcaRuntimeMetrics.ACTIONS_PUBLISHED, action.name(), 1);
+            if (action.impactedNodes() != null) {
+                final String nodeIds = action.impactedNodes().stream()
+                        .map(n -> n.getNodeId().toString())
+                        .collect(Collectors.joining(",", "{", "}"));
```

Comment:
I am thinking if we should just keep it simple and join them with comma as delimiter ?

Reply:
Done